packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -32,12 +110,16 @@ const TreeNode = React.forwardRef(
       onToggle,
       onTreeSelect,
       renderIcon: Icon,
-      selected,
+      selected: propSelected,
       value,
       ...rest
     },
     ref
   ) => {
+    // These are provided by the parent TreeView component
+    const depth = propDepth as number;
+    const selected = propSelected as (string | number)[];
```

Comment:
I have another proposal.

Since the component internals expect that these should always be defined, should we do something like this to provide a better error for consumers that try to mount TreeNode outside of TreeView?

```typescript
if (depth === undefined || selected === undefined){
   throw new Error("It looks like you are trying to mount a TreeNode outside of a TreeView. To do this, ensure that the 'selected' and 'depth' props are provided.");
}
```

This would also mean that we wouldn't have to cast the types manually either.